pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -272,6 +274,23 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 	// Reaching this point, we had an error in processing, clearing up global error
 	podAutoscalerInternal.SetError(nil)
 
+	// Check that targetRef is not set to the cluster agent
+	clusterAgentRef := autoscalingv2.CrossVersionObjectReference{
+		Kind:       "Deployment",
+		Name:       getDeploymentName(),
+		APIVersion: "apps/v1",
+	}
+
+	if podAutoscaler.Spec.TargetRef == clusterAgentRef {
```

Comment:
nit: The code should probably go to a dedicated function like `validateAutoscaler` or `validateRequirements`, the max 100 could also go there.